packages/graphql/src/translate/where/property-operations/create-connection-operation.ts

```diff
@@ -56,13 +56,31 @@ export function createConnectionOperation({
     const operations: (Cypher.BooleanOp | Cypher.RawCypher | undefined)[] = [];
 
     Object.entries(nodeEntries).forEach((entry) => {
-        const refNode = context.nodes.find(
-            (x) => x.name === entry[0] || x.interfaces.some((i) => i.name.value === entry[0])
-        ) as Node;
+        let nodeOnValue: string | undefined = undefined;
+        const nodeOnObj = entry[1]?.node?._on;
+        if (nodeOnObj) {
+            nodeOnValue = Object.keys(nodeOnObj)[0];
+        }
+
+        let refNode = context.nodes.find((x) => x.name === nodeOnValue || x.name === entry[0]) as Node;
+        if (!refNode) {
+            refNode = context.nodes.find((x) => x.interfaces.some((i) => i.name.value === entry[0])) as Node;
+        }
 
         const relationField = connectionField.relationship;
 
-        const childNode = new Cypher.Node({ labels: refNode.getLabels(context) });
+        let labels = refNode.getLabels(context);
+        const hasOnlyNodeObjectFilter = entry[1]?.node && !nodeOnObj;
+        if (hasOnlyNodeObjectFilter) {
+            const nodesImplementingInterface = context.nodes.filter((x) =>
+                x.interfaces.some((i) => i.name.value === entry[0])
+            );
+            if (nodesImplementingInterface.length) {
+                // remove the labels so we check all possibilities and not just the one label of the refNode
+                labels = [];
```

Comment:
I'm not sure this is right. Do we not need to have an OR of all the types of the interface as you suggested?

e.g.

```
MATCH (this:`Film`)
WHERE size([(this1:`Dishney`|`Netflix`|`Prime`)-[this0:DISTRIBUTED_BY]->(this) WHERE this1.name = $param0 | 1]) > 0
RETURN this { .title } AS this
```

It probably is a bit of an edge case but it would prevent matching nodes of a different type that share the same relationship. For example, with this type def it could end up also matching a `Publisher` if there are no labels as they also use the `DISTRIBUTED_BY` relationship type:

```gql
type Movie implements Production @node(label: "Film") {
    title: String!
    actors: [Actor!]! @relationship(type: "ACTED_IN", direction: IN, properties: "ActedIn")
    runtime: Int!
    distribution: [DistributionHouse!]! @relationship(type: "DISTRIBUTED_BY", direction: IN)
    publisher: Publisher! @relationship(type: "DISTRIBUTED_BY", direction: IN)
}
```

Reply:
It's good that you know or find these edge cases!
This approach with using an OR expression, `(this1:Dishney|Netflix|Prime)`, is sadly not Neo4j 4.x safe but there is another way to solve it. It's however a bit more verbose.

Reply:
Sorry to keep adding comments to this! Is it possible to add an integration test for this case as it seems like something that call be missed by changes in the future?